Methane_MarkDown.Rmd

```diff
@@ -111,3 +110,206 @@ ggplot(data= results, aes(x= year, y= value, color= units))+
          color = "Units") + facet_wrap(~variable)
 ```
 ... to show our data side-by-side.
+
+## Sensitivity Analysis
+
+To examine SSP245's Natural methane  sensitivity we first ask hector to fetch data for the `NATURAL_CH4` constant. We then create a function to run this parameter set to a particular value. 
+
+```{r call-for-param}
+core <- newcore(ini_file)
+run(core)
+default<- fetchvars(core, NA, NATURAL_CH4())
+
+param<- function(core, parameter, value) {
+  old_value <- fetchvars(core, NA, parameter)
+  unit <- as.character(old_value[["units"]])
+  setvar(core, NA, parameter, value, unit)
+  reset(core)
+  run(core)
+  result <- fetchvars(core, 2000:2100)
+  result[["parameter_value"]] <- value
+  result
+}
+```
+
+Then we tell hector to run a range of values. In this case we ask for 20% above and bellow the set value of `NATURAL_CH4`(341Tg) at intervals of 5 +/-20%. These set values are labelled as `ch4-seq`.
+
+```{r sen-param}
+run_with_param_range <- function(core, parameter, values) {
+  mapped <- Map(function(x) param(core, parameter, x), values)
+  Reduce(rbind, mapped)
+}
+valuen<- default %>%
+  select(value)
+meann<- valuen$value
+ch4_low <- meann * (1 - 0.2)
+ch4_high <- meann * (1 + 0.2)
+ch4_seq <- seq(ch4_low, ch4_high, 5)
+sensitivity_emission <- run_with_param_range(core, NATURAL_CH4(), ch4_seq)
+```
+
+Once we have our differing model outputs we can create a graph. Using the graph function `facet_wrap`to separate the results based on variable. 
+
+```{r sen-graph}
+ggplot(sensitivity_emission) +
+  aes(x = year, y = value, color = parameter_value, group = parameter_value) +
+  geom_line() +
+  facet_wrap(~variable, scales = "free_y") +
+  labs(color = "NATURAL CH4 (Tg)") +
+  scale_color_viridis_c() 
+```
+
+### Adding a Histogram 
+
+Using the same `Natural_CH4` data, we can also make a histogram plot. Since we already made `ch4_low` and `ch4_high` with levels +/-20%, we can re-purpose them for this graph.
+
+```{r rnorm1}
+sdn<- meann*0.2
+data<- data.frame(value= rnorm(1000, meann, sdn))
+```
+
+Then we can use `ggplot2` to create a histogram with the randomized values.
+```{r hist}
+ggplot(data, aes(x=value))+
+  geom_histogram(binwidth = 1)+
+  labs( x= "Natural CH4 Level", y= "count", main= "Natural CH4 Histogram")
+invisible(shutdown(core))
+```
+
+
+## Multiple Parameter Run
+
+Working with four parameters `NATURAL_CH4`, `PREINDUSTRIAL_CH4`, `LIFETIME_SOIL` and `LIFETIME_STRAT` to create a sensitivity analysis. First we run all of them through `fetchvars`.
+
+```{r core}
+core <- newcore(ini_file)
```

Comment:
If you don't shutdown the core above, you can just reset it here and use the same core as above